include/bitcoin/bitcoin/utility/random.hpp

```diff
@@ -50,10 +57,52 @@ BC_API void pseudo_random_fill(data_chunk& chunk);
  * @param[in]  ratio    The determinant of the minimum duration as the inverse
  *                      portion of the maximum duration.
  * @return              The randomized duration.
+ *
+ * DEPRECATED in v3
  */
 BC_API asio::duration pseudo_randomize(const asio::duration& maximum,
     uint8_t ratio=2);
 
+
+class pseudo_random
+{
+  public:
+    /**
+     * Returns a thread-safe static random generator instance.
+     * @return  The generator.
+     */
+    static std::mt19937& get_twister();
+
+    /**
+     * Generate a pseudo random number within the domain.
+     * @return  The 64 bit number.
+     */
+    static BC_API uint64_t next();
```

Comment:
The `BC_API` should qualify the class (except for template classes, which are embedded, not linked).